### PR TITLE
Collect entity from parser, support Entity.Strict and Empty

### PR DESCRIPTION
**Pull request by wjoel**
Together with the other PRs (#6080, #6091, #6092), this increases performance on "plaintext" from the Techempower benchmarks by around 125% (from 172k reqs/s to 392k reqs/s), and around 80% for JSON.

### CODE REVIEW

**Review by rossabaker — APPROVED** [review]

File: blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala [hunk 1]
```diff
@@ -405,7 +405,12 @@ private final class Http1Connection[F[_]](
           status = status,
           httpVersion = httpVersion,
           headers = headers,
-          entity = Entity(body.interruptWhen(idleTimeoutS)),
+          entity = entity match {
+            case Entity.Default(body, length) =>
+              Entity[F](body.interruptWhen(idleTimeoutS), length)
+            case _ =>
```
Comment (danicheg) [inline blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala:411]: Even it's hard to imagine we will add another subtype of `Entity`, nevertheless, I'm sure that using a wildcard on pattern matching isn't good practice. Verbosity isn't always bad.
```suggestion
            case Entity.Strict(_) | Entity.Empty =>
```

Comment (wjoel) [reply blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala]: Good point, thanks.

**Comment by wjoel** [comment]
Flaky test?

**Comment by armanbilge** [comment]
Yep, we have loads. Feel free to ignore :)

**Review by ChristopherDavenport — APPROVED** [review]

**Comment by danicheg** [comment]
I think this is ready for merge. But let's try to achieve the 'green CI' (just for fun, you know).

**Pull request merged**